connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```
@@ -105,6 +108,49 @@ public void stop() {
+                    .findAny()
+                    .orElseGet(() -> {
+                        ConfigValue result = new ConfigValue(MirrorCheckpointConfig.SYNC_GROUP_OFFSETS_ENABLED);
+                        configValues.add(result);
+                        return result;
+                    });
+
+            ConfigValue emitCheckpoints = configValues.stream().filter(prop -> MirrorCheckpointConfig.EMIT_CHECKPOINTS_ENABLED.equals(prop.name()))
+                    .findAny()
+                    .orElseGet(() -> {
+                        ConfigValue result = new ConfigValue(MirrorCheckpointConfig.EMIT_CHECKPOINTS_ENABLED);
+                        configValues.add(result);
+                        return result;
+                    });
+
+            String errorMessage = "MirrorCheckpointConnector can't run with both" +
+                    MirrorCheckpointConfig.SYNC_GROUP_OFFSETS_ENABLED + ", " + MirrorCheckpointConfig.EMIT_CHECKPOINTS_ENABLED + "set to false";
+            syncGroupOffsets.addErrorMessage(errorMessage);
+            emitCheckpoints.addErrorMessage(errorMessage);
+        }
+        if ("false".equals(emitOffsetSyncsValue) && ("true".equals(emitCheckpointsValue) || "true".equals(syncGroupOffsetsValue))) {
+            ConfigValue emitOffsetSyncs = configValues.stream().filter(prop -> MirrorConnectorConfig.EMIT_OFFSET_SYNCS_ENABLED.equals(prop.name()))
+                    .findAny()
+                    .orElseGet(() -> {
+                        ConfigValue result = new ConfigValue(MirrorConnectorConfig.EMIT_OFFSET_SYNCS_ENABLED);
+                        configValues.add(result);
+                        return result;
+                    });
+
+            emitOffsetSyncs.addErrorMessage("MirrorCheckpointConnector can't run while MirrorSourceConnector configured with" +
+                    MirrorConnectorConfig.EMIT_OFFSET_SYNCS_ENABLED + "set to false");
+        }
+        return new Config(configValues);
```

Can these validations live in the config class? In this case that would be `MirrorConnectConfig`. 

Besides being in line with config validation it has the added benefit that we can use `.getBoolean(...)` instead of ` props.get(..prop.)).orElse(Boolean.toString(...default...)`.

---

> Can these validations live in the config class? In this case that would be MirrorConnectConfig.

`MirrorConnectConfig` is more generic and shared between multiple connectors. We shouldn't move this validation there maybe `MirrorCheckpointConfig` but not `MirrorConnectConfig`

> Besides being in line with config validation it has the added benefit that we can use .getBoolean(...) instead of  props.get(..prop.)).orElse(Boolean.toString(...default...).

I think we should to leverage `Connector::validate` here as it get called before `start` and we can fail faster. If we want to use `getBooleans` maybe we should initialise `MirrorCheckpointConfig` within the validate but we shouldn't move this from `Connector::validate`

---

Just want to clarify some points regarding connect (to the extend of my understanding):
1. `Connector::validate` get called by the runtime before `Connector::start` which will get us to fail faster
2. `Connector::validate` get called in `AbstractHerder::validateConnectorConfig` which handles connector/task lifecycle tracking. This also get called from connect REST api `config/validate`
3. While `Connector::start` get called during the reconfiguration of the plugin in connect as well as at the start

So we can't move the validation to `Connector::start` and we can't move the initialisation of `MirrorConnectorConfig` into `Connector::validate`. 

If we want to use `.getBoolean` then I would suggest having a look into my last [commit](https://github.com/apache/kafka/pull/15999/commits/27c07546e6b6983e79b1c5989564b337f84e930c) where I temporary initialised a connect config to use the getters from `AbstractConfig` which all mirror connector configs extends.

---

Thanks for the explanation Omnia. That makes sense. Could `MirrorCheckpointConfig` have a new method performs these validations, and we call that method from `MirrorCheckPointconnector#validate`? Maybe the method could return `Optional<ConfigValue>` if there is any validation issue?